src/OrchardCore.Modules/OrchardCore.Users/ConfigureCookieAuthenticationOptions.cs

```diff
@@ -0,0 +1,36 @@
+using Microsoft.AspNetCore.Authentication.Cookies;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+
+using OrchardCore.Settings;
+using OrchardCore.Users.Controllers;
+
+namespace OrchardCore.Users
+{
+    public class ConfigureCookieAuthenticationOptions : IConfigureNamedOptions<CookieAuthenticationOptions>
+    {
+        private readonly IServiceScopeFactory _serviceScopeFactory;
+
+        public ConfigureCookieAuthenticationOptions(IServiceScopeFactory serviceScopeFactory)
+        {
+            _serviceScopeFactory = serviceScopeFactory;
+        }
+
+        public void Configure(CookieAuthenticationOptions options)
+        {
+            Configure(null, options);
+        }
+
+        public void Configure(string name, CookieAuthenticationOptions options)
+        {
+            using (var scope = _serviceScopeFactory.CreateScope())
+            {
+                var settings = scope.ServiceProvider.GetRequiredService<ISiteService>().GetSiteSettingsAsync().Result;
+                var baseUrl = settings.BaseUrl?.TrimEnd('/');
+                options.LoginPath = baseUrl + "/" + nameof(AccountController.Login);
+                options.AccessDeniedPath = baseUrl + "/Error/403";
+                options.LogoutPath = baseUrl + "/" + nameof(AccountController.LogOff);
+            }
```

Comment:
So here we will always override what we already do in `services.ConfigureApplicationCookie()` and that take into account the `UserOptions` coming from the configuration (e.g. appsettings.json). So, We would need only one `CookieAuthenticationOptions` configuration registered through the `OC.Users` startup and that still uses `UserOptions`.

With `UserOptions` we already can override the `LoginPath` and the `LogoffPath` (but only used in a `MapAreaControllerRoute()`, we never set the `LogoutPath` option), the missing one is the `AccessDeniedPath`, so maybe add this option in `UserOptions`, and / or add the option to use the base url for these paths.

**But TBH i don't think it is good to do all of this, maybe what you are trying to do could be fixed by just using the `OC.ReverseProxy` module**

Reply:
I've made the PR when `UserOptions` did not exist.

> **But TBH i don't think it is good to do all of this, maybe what you are trying to do could be fixed by just using the `OC.ReverseProxy` module**

You are not paying attention to the problem I'm trying to resolve. 
**Reverse proxy has nothing to do with configured cookie, authentication handler and login urls**

1.  The site url must be used **everywhere**
2. The base path must be considered **everywhere**

TBH I've already made hacks to my system like `app.UsePathBase("/cms");` because I need the CMS actually running and have no time arguing. 
This is such an essential setting that should work that I don't know **any** CMS/CMF that has this setting lacking or not working as intended